server/api/impl/aws/provision_app.go

```diff
@@ -187,25 +219,21 @@ func (c *Client) createFunction(zipFile io.Reader, function, handler, runtime, r
 	return nil
 }
 
-// getFunctionName generates function name for a specific app
-// name can be 64 characters long.
-func getFunctionName(appID apps.AppID, version apps.AppVersion, function string) (string, error) {
-	if len(appID) > appIDLengthLimit {
-		return "", errors.Errorf("appID %s too long, should be %d bytes", appID, appIDLengthLimit)
-	}
-	if len(version) > len(versionFormat) {
-		return "", errors.Errorf("version %s too long, should be in %s format", version, versionFormat)
-	}
-	name := fmt.Sprintf("%s-%s-%s", appID, version, function)
-	if len(name) <= lambdaFunctionFileNameMaxSize {
-		return name, nil
+func (c *Client) provisionAssets(manifest *apps.Manifest, assets []assetData) (*apps.Manifest, error) {
+	if manifest.Assets == nil {
+		manifest.Assets = make([]apps.Asset, 0, len(assets))
 	}
-	functionNameLength := lambdaFunctionFileNameMaxSize - len(appID) - len(version) - 2
-	hash := sha256.Sum256([]byte(name))
-	hashString := hex.EncodeToString(hash[:])
-	if len(hashString) > functionNameLength {
-		hashString = hashString[:functionNameLength]
-	}
-	name = fmt.Sprintf("%s-%s-%s", appID, version, hashString)
-	return name, nil
+	for _, asset := range assets {
+		key := getAssetFileKey(manifest.AppID, manifest.Version, asset.name)
+		if err := c.S3FileUpload(key, asset.file); err != nil {
+			return nil, errors.Wrapf(err, "can't provision asset - %s with key - %s", asset.name, key)
+		}
+		manifest.Assets = append(manifest.Assets, apps.Asset{
+			Name:   asset.name,
+			Type:   apps.S3Asset,
+			Bucket: c.appsS3Bucket,
+			Key:    key,
+		})
```

Comment:
It would be simpler if the manifest would be immutable and we won't store that information. Can we generate the information on the fly? We only have one asset type and the bucket should be the same for all apps.